WMF Framework/EventPlatformClient/EventPlatformClient.swift

```diff
@@ -263,9 +255,12 @@ public class EPC: NSObject {
 
         self.streamIntakeServiceURI = streamIntakeServiceURI
         self.streamConfigServiceURI = streamConfigServiceURI
-        
+
         super.init()
 
+        self.samplingController.appInstallId = installID
```

Comment:
I'm a bit wary of `SamplingController` having its own copies of these values, since they could easily get out of sync (such as when `resetSession` is called from this class). I think better would be for SamplingController to have EventPlatformClientLibrary as a delegate, then whenever SamplingController needs those identifiers it asks the delegate (I worked something up right quick [here](https://github.com/wikimedia/wikipedia-ios/commit/530075c4b964d1a0d2e1b8405ea4158d50a557ac) if you want to reference it).

Reply:
Thanks for this.   Done.